test/invariant/Lockup.t.sol

```diff
@@ -95,6 +95,15 @@ abstract contract Lockup_Invariant_Test is Invariant_Test {
         }
     }
 
+    function invariant_DepositedAmountNotZero() external useCurrentTimestamp {
+        uint256 lastStreamId = lockupStore.lastStreamId();
+        for (uint256 i = 0; i < lastStreamId; ++i) {
+            uint256 streamId = lockupStore.streamIds(i);
+            uint128 depositAmount = lockup.getDepositedAmount(streamId);
+            assertNotEq(depositAmount, 0, "Invariant violated: stream non-null, deposited amount zero");
```

Comment:
A thought:

The `depositAmount` and `startTime` are set at the time of creating streams. They are never modified throughout the lifetime of the stream. I don't think they need to be tested as invariants 🤔. Of course we can compare them against zero, but is that a good invariant? If yes, then why not other variables such as `segmentCount` etc.?

Reply:
We absolutely should keep these tests.

> They are never modified throughout the lifetime of the stream

That is an invariant - the point of an invariant test is to prove that the invariant holds.

It all boils down to fallibilism - an unknown bug could set the deposit amount to zero, or the start time to zero, during the lifecycle of the stream.

> then why not other variables such as segmentCount etc.?

Maybe we should test other variables, too!

But not `segmentCount` because that's not part of contract storage. It's a variable dynamically generated at runtime, and it's associated with a specific function.

Reply:
Fair and agree. Thanks for your comment.

Reply:
No prob